src/System.Threading.Channels/src/System/Threading/Channels/AsyncOperation.cs

```diff
@@ -25,62 +27,117 @@ internal abstract class ResettableValueTaskSource
+        /// <remarks>
+        /// If the operation is cancelable, then it can't be pooled.  And if it's poolable, there must never be race conditions to complete it,
+        /// which is the main reason poolable objects can't be cancelable, as then cancellation could fire, the object could get reused,
+        /// and then we may end up trying to complete an object that's used by someone else.
+        /// </remarks>
+        private readonly bool _pooled;
+        /// <summary>Whether continuations should be forced to run asynchronously.</summary>
+        private readonly bool _runContinuationsAsynchronously;
+
+        /// <summary>Only relevant to cancelable operations; 0 if the operation hasn't had completion reserved, 1 if it has.</summary>
+        private volatile int _completionReserved = 0;
+        /// <summary>The result of the operation.</summary>
+        private TResult _result;
+        /// <summary>Any error that occurred during the operation.</summary>
         private ExceptionDispatchInfo _error;
+        /// <summary>The continuation callback.</summary>
+        /// <remarks>
+        /// This may be the completion sentinel if the operation has already completed.
+        /// This may be the available sentinel if the operation is being pooled and is available for use.
+        /// This may be null if the operation is pending.
+        /// This may be another callback if the operation has had a callback hooked up with OnCompleted.
+        /// </remarks>
         private Action<object> _continuation;
+        /// <summary>State object to be passed to <see cref="_continuation"/>.</summary>
         private object _continuationState;
+        /// <summary>Scheduling context (a <see cref="SynchronizationContext"/> or <see cref="TaskScheduler"/>) to which to queue the continuation. May be null.</summary>
         private object _schedulingContext;
+        /// <summary>Execution context to use when invoking <see cref="_continuation"/>. May be null.</summary>
         private ExecutionContext _executionContext;
+        /// <summary>The token value associated with the current operation.</summary>
+        /// <remarks>
+        /// IValueTaskSource operations on this instance are only valid if the provided token matches this value,
+        /// which is incremented once GetResult is called to avoid multiple awaits on the same instance.
+        /// </remarks>
         private short _currentId;
 
+        /// <summary>Initializes the interactor.</summary>
+        /// <param name="runContinuationsAsynchronously">true if continuations should be forced to run asynchronously; otherwise, false.</param>
+        /// <param name="cancellationToken">The cancellation token used to cancel the operation.</param>
+        /// <param name="pooled">Whether this instance is pooled and reused.</param>
+        public AsyncOperation(bool runContinuationsAsynchronously, CancellationToken cancellationToken = default, bool pooled = false)
```

Comment:
>pooled [](start = 119, length = 6)

do we really need pooled param? can we assume all non cancelable token can be pooled?